src/mp/math_utils.py

```diff
@@ -41,4 +42,4 @@ def get_distinct_prime_facs(num):
 
 # One-liner fibonacci for a record
 def fib(n):
-    return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n-1), (1, 1))[0]
+    return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n), (0, 1))[1]
```

Comment:
Beautiful functional fibonacci.
I think you meant to leave the [0] as [0] though so
`return reduce(lambda item, _: (item[1], item[0] + item[1]), range(n), (0, 1))[0]`

Reply:
Yea, people seem to use different conventions, so if it starts with fib(0) = 1 or 0 makes difference sometimes...